vpr/src/route/router_lookahead_map.cpp

```diff
@@ -215,37 +243,76 @@ float get_lookahead_map_cost(int from_node_ind, int to_node_ind, float criticali
     auto& device_ctx = g_vpr_ctx.device();
     auto& from_node = device_ctx.rr_nodes[from_node_ind];
 
-    e_rr_type from_type = from_node.type();
-    int from_cost_index = from_node.cost_index();
-    int from_seg_index = device_ctx.rr_indexed_data[from_cost_index].seg_index;
-
-    VTR_ASSERT(from_seg_index >= 0);
-
     int delta_x, delta_y;
     get_xy_deltas(from_node_ind, to_node_ind, &delta_x, &delta_y);
     delta_x = abs(delta_x);
     delta_y = abs(delta_y);
 
-    /* now get the expected cost from our lookahead map */
-    int from_chan_index = 0;
-    if (from_type == CHANY) {
-        from_chan_index = 1;
-    }
+    float expected_cost = std::numeric_limits<float>::infinity();
+
+    e_rr_type from_type = from_node.type();
+    if (from_type == SOURCE || from_type == OPIN) {
```

Comment:
Can you explain what is going on with this change?  It seems orthogonal to the overall PR?

Reply:
In order to get reasonable delay estimates before the 1st routing iteration (to figure out criticalities) we use the map lookahead. However the map lookahead (before these changes) would crash if you tried to estimate delay from something other than a CHANX/CHANY.

This extends the map lookahead to behave reasonably for estimating delays from SOURCEs (which is used for pre-routing delay estimates). See comments with e2551a5 and 04f7faf.

Reply:
Comments in the commits are useful, but what you just wrote really ought to be in the code.

Reply:
Agreed I'll put some more comments in.